### ISSUE TITLE
Reproduction of results

### ISSUE THREAD

**Issue by eliocamp** [richarddmorey/Morey_Hoekstra_StatCognition#1]
Hi! 

I participated of RepoHack and chose this paper to reproduce. I filled out the author feedback form, but I also took notes documenting my experience and some more detailed information which might be useful to you. 

## General notes

This is such a great repo to work with. I was incredibly impressed at the work you put into making your research as open and easy to reproduce as possible. Awesome stuff include:
 - a video showcasing the experimental task 
 - a webpage where anyone can demo the experimental task
 - a webpage to see and filter the actual data! 🤩

There's a lot of documentation, both of functions and datasets, which I'm sure will make it very easy to go deep into the code and run alternative analysis. 

Maybe I missed it, but I don't see a version of the manuscript rendered with the current version of the repo. This would've been useful in order to compare my resulting pdf with the "correct" or submitted one. 
  
I had some issues with some dependencies and had one error that forced me to modify the underlying code. Fortunately, it was mostly aesthetic. 

The actual manuscript is clear and celebrate the heavy use of inline R code to prevent any typo or error create by copy-pasting numbers from one place to another. As a small detailT the functions that create figures (i.e. `fig_error_rates()`) read de data hard-coded in the package, so there's no easy way to reuse the same code to new data if a researcher wanted to replicate the experiment. 

Adding unit testing might help with robustness. As far as I saw, there isn't any checking of arguments or results. It's not critical and it's a lot of work, but it's still good practice.

All in all, this is a great example of reproducible research. As a minor issue, there are some warnings. One comes from from using a deprecated function (which probably got deprecated after the paper was published) and another mistery one during installation. 

The only mayor issue are the dependencies. They don't seem to be listed correctly as I needed to install several before being able to get the .Rmd to compile. Some dependencies listed as Import are not needed to run the analysis (such as shiny, or htmlwidgets) and others listed in Suggest need to be installed to get the correct output (such as kableExtra and R.rsp). 

This is an interesting paper and a very well put together repository. The fact that such a perfect package still took me some effort to reproduce, really opened my eyes at how hard it is to make research completely reproducible. I applaud your work!

## Documented process:

sessionInfo(): 
```
R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04 LTS
```

1. I forked the repo and cloned it to my machine,

2. Installed dependencies with `devtools::install_deps()`.

3. Tried to install the package (R CMD INSTALL, using devtools) and ran into missing dependenies (see below) which I installed as they appeared.

4. After successfully installing the  package, I tried `explore_data()` and `task_demo()`. They worked flawlessly! 

5. Tried to build the vignettes (via RStudio Crtl + Shift +D) and ran into other missing dependencies (see below). Also, had Warning #1  (not sure what that is)

5. After installing the dependencies, I ran into Error #1.

6. Ran `devtools::build_vignettes()`, which started to automatically install 5 missing packages.

7. After installing those dependencies, I still got Error #1 and now also Error #2 and #3.   
  I also got Warnings #2 and #3.

8. Tried with `devtools::install_github("richarddmorey/Morey_Hoekstra_StatCognition", build_vignettes = TRUE, dependencies = TRUE)` and ran into Error #1 and #2 and #3.

9. Decided to ran a CRAN check with `detools::check()`. It failed and halted do to Error #1.

10. Changed `papaja::apa_print(kt_expt)$statistic` to `round(kt_expt$statistic, 2)` in line 262 of supplementA.Rmd and rebuild vignettes. Success! 🎉

11. Ran `devtools::check()` again. Got Error #4 (nothing too serious)

## Missing Dependencies:

These dependencies are needed to compile the manuscript but they are listed in Suggests:

 - rticles
 - rmdformats
 - kableExtra
 - R.rsp

(This list might not be complete, since they are only the ones I didn't have already installed)

Other packages don't seem to be required to compile the manuscript but are listed in Imports. These include shiny, servr, htmwidgets, DT

## Warnings

1. `It is recommended to use ‘given’ instead of ‘middle’.`

2. `Warning: R.methodsS3::throw() is deprecated. Use base::stop() instead, or R.oo::throw().`

3. 
```
E> In tools::buildVignettes(dir = ".", tangle = TRUE) :
E>   Files named as vignettes but with no recognized vignette engine:
E>    ‘vignettes/data_access.Rmd’
E>    ‘vignettes/design_thoughts.Rmd’
E>    ‘vignettes/man_supp.Rmd’
E>    ‘vignettes/task_demo.Rmd’
E>    ‘vignettes/task_materials.Rmd’
```

## Errors

1. Error on line 262 of supplementA.Rmd (specifically, the inline R expression `papaja::apa_print(kt_expt)$statistic`): `no applicable method for 'apa_print' applied to an object of class "htest"`

2. Error: processing vignette 'supplementA.html.asis' failed with diagnostics:
   No such output file: supplementA.html

3. Error: processing vignette 'supplementB.html.asis' failed with diagnostics:
   No such output file: supplementB.html
   
4. Namespace dependency not required: ‘gridExtra’   
    Imports includes 21 non-default packages.    
  Importing from so many packages makes the package vulnerable to any of    
  them becoming unavailable.  Move as many as possible to Suggests and    
  use conditionally.

**Comment by richarddmorey** [comment]
Thanks for this! I'll look into the issues.

**Comment by richarddmorey** [comment]
Re: importans and suggests, I think I have done everything according to the R package development guidelines (which may or may not be the best for easily allowing someone to manually reproduce the R code). Things required for "vignettes" (of which the manuscript is an example) should be put in Suggests, and aren't necessarily automatically installed with the package. Some of the things in "Imports" are required for the data exploration interface.

If you reproduce the document via `R CMD build`, building the vignettes, I think it will install the Suggests. Maybe a "guide to reproducing the manuscript" would be something I can add to the vignette, to clarify these issues?

**Comment by richarddmorey** [comment]
> Error on line 262 of supplementA.Rmd (specifically, the inline R expression papaja::apa_print(kt_expt)$statistic): no applicable method for 'apa_print' applied to an object of class "htest"

I cannot reproduce this one. What version of `papaja` are you running, if you don't mind me asking?

**Comment by eliocamp** [comment]
I installed the version on github at the time. It's 0.1.0.9942. Funny that you couldn't reproduce my reproducibility report 😂️

Re: packages, maybe the issue is that CRAN think of vignettes as optional pieces of documentation that are not crucial to the normal functioning of the package. Thus, it makes sense to relegate vignette building to Suggests. But if the main functionality of a package is to create a report, then the priorities should change? In any case, I think that following CRAN recommendations is fair enough.

**Comment by richarddmorey** [comment]
>  In any case, I think that following CRAN recommendations is fair enough.

Yeah, I'm not sure, given that I'll never submit to CRAN. My real goal is ease of reproducibility -- and also not getting unnecessarily emails from people who can't reproduce the results :) -- so that should probably guide things. 

Thanks to your report, though, I did catch some things that changed in R 3.6+ and newer versions of pandoc that kept the document from compiling. We're resubmitting soon, so the report was well-timed. Thanks!

**Comment by eliocamp** [comment]
I'm glad it helped :D 
It was my first time trying to reproduce other's people results int this fashion.

**Comment by richarddmorey** [comment]
> Error on line 262 of supplementA.Rmd (specifically, the inline R expression papaja::apa_print(kt_expt)$statistic): no applicable method for 'apa_print' applied to an object of class "htest"

I take it back; I just upgraded to R 4.0 and after installing all the new packages, I now get this error.

**Comment by RinkHoekstra** [comment]
Hi Richard,
 in case we are waiting for each other: I’d like to take a look at the latest version of our paper. Could you send a link? Or is it in Dropbox? I’d like to be sure I have the right version. What’s the proposed outlet?
Best,
Rink

**Comment by richarddmorey** [comment]
This seems to be resolved now, with sometime after R 4.0; closing up.
